core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/deletevolumes/DeleteVolumesActions.java

```diff
@@ -0,0 +1,229 @@
+                            && resource.getResourceType().equals(ResourceType.AWS_VOLUMESET))
+                        .map(s -> cloudResourceConverter.convert(s)).collect(toList());
+                    String cloudPlatform = stack.getCloudPlatform();
+                    deleteVolumesService.stopClouderaManagerService(stack, hostTemplateServiceComponents);
+                    DeleteVolumesRequest deleteVolumesRequest = new DeleteVolumesRequest(cloudResourcesToBeDeleted, stackDeleteVolumesRequest, cloudPlatform,
+                            hostTemplateServiceComponents);
+                    sendEvent(context, DELETE_VOLUMES_EVENT.selector(), deleteVolumesRequest);
+                }
+            }
+        };
+    }
+
+    @Bean(name = "DELETE_VOLUMES_FINISHED_STATE")
+    public Action<?, ?> deleteVolumesFinishedAction() {
+        return new AbstractClusterAction<>(DeleteVolumesRequest.class) {
+
+            @Override
+            protected void doExecute(ClusterViewContext context, DeleteVolumesRequest payload, Map<Object, Object> variables) {
+                StackDeleteVolumesRequest stackDeleteVolumesRequest = payload.getStackDeleteVolumesRequest();
+                StackDto stack = stackDtoService.getById(payload.getResourceId());
+                String requestGroup = stackDeleteVolumesRequest.getGroup();
+                String cloudPlatform = payload.getCloudPlatform();
+                CloudPlatformVariant cloudPlatformVariant = new CloudPlatformVariant(Platform.platform(cloudPlatform), Variant.variant(cloudPlatform));
+                CloudConnector cloudConnector = cloudPlatformConnectors.get(cloudPlatformVariant);
+                CloudCredential cloudCredential = stackUtil.getCloudCredential(stack.getEnvironmentCrn());
+                Location location = location(region(stack.getRegion()), availabilityZone(stack.getAvailabilityZone()));
+                CloudContext cloudContext = CloudContext.Builder.builder()
+                        .withId(stack.getId())
+                        .withName(stack.getName())
+                        .withCrn(stack.getResourceCrn())
+                        .withPlatform(stack.getCloudPlatform())
+                        .withVariant(stack.getPlatformVariant())
+                        .withLocation(location)
+                        .withWorkspaceId(stack.getWorkspace().getId())
+                        .withAccountId(Crn.safeFromString(stack.getResourceCrn()).getAccountId())
+                        .build();
+                AuthenticatedContext ac = getAuthenticatedContext(cloudCredential, cloudContext, cloudConnector);
+                List<CloudResource> cloudResourcesToBeDeleted = payload.getResourcesToBeDeleted();
+                Set<ServiceComponent> hostTemplateServiceComponents = payload.getHostTemplateServiceComponents();
+                try {
+                    deleteVolumesService.detachResources(cloudResourcesToBeDeleted, cloudPlatform, ac);
```

Comment:
The same here(individual handler to eliminate the business logic from the action) and not to mention that the result of the detach/delete is not really considered before starting the CM services.
What happens if a disk remains on the provider side or silently couldn't be detached and deleted?

Reply:
done